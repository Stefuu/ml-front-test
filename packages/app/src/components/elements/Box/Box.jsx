import React, { Children } from 'react'
import styled from 'styled-components'
import {
  alignContent,
  alignItems,
  alignSelf,
  background,
  backgroundImage,
  backgroundPosition,
  backgroundRepeat,
  backgroundSize,
  border,
  borderBottom,
  borderLeft,
  borderRadius,
  borderRight,
  borderTop,
  bottom,
  color,
  display,
  flex,
  flexBasis,
  flexDirection,
  flexWrap,
  fontFamily,
  fontSize,
  fontWeight,
  height,
  justifyContent,
  justifyItems,
  justifySelf,
  left,
  lineHeight,
  maxHeight,
  maxWidth,
  minHeight,
  minWidth,
  opacity,
  order,
  overflow,
  position,
  right,
  size,
  space,
  textAlign,
  top,
  width,
  zIndex
} from 'styled-system'
import tag from 'clean-tag'

const Box = styled(({ children, ...props }) => <tag.div {...props}>{Children.map(children, child => child)}</tag.div>)`
  ${alignContent}
  ${alignItems}
  ${alignSelf}
  ${background}
  ${backgroundImage}
  ${backgroundPosition}
  ${backgroundRepeat}
  ${backgroundSize}
  ${border}
  ${borderBottom}
  ${borderLeft}
  ${borderRadius}
  ${borderRight}
  ${borderTop}
  ${bottom}
  ${color}
  ${display}
  ${flex}
  ${flexBasis}
  ${flexDirection}
  ${flexWrap}
  ${fontFamily}
  ${fontSize}
  ${fontWeight}
  ${height}
  ${justifyContent}
  ${justifyItems}
  ${justifySelf}
  ${left}
  ${lineHeight}
  ${maxHeight}
  ${maxWidth}
  ${minHeight}
  ${minWidth}
  ${opacity}
  ${order}
  ${overflow}
  ${position}
  ${right}
  ${size}
  ${space}
  ${textAlign}
  ${top}
  ${width}
  ${zIndex}
`

Box.propTypes = {
  ...alignContent.propTypes,
  ...alignItems.propTypes,
  ...alignSelf.propTypes,
  ...background.propTypes,
  ...backgroundImage.propTypes,
  ...backgroundPosition.propTypes,
  ...backgroundRepeat.propTypes,
  ...backgroundSize.propTypes,
  ...border.propTypes,
  ...borderBottom.propTypes,
  ...borderLeft.propTypes,
  ...borderRadius.propTypes,
  ...borderRight.propTypes,
  ...borderTop.propTypes,
  ...bottom.propTypes,
  ...color.propTypes,
  ...display.propTypes,
  ...flex.propTypes,
  ...flexBasis.propTypes,
  ...flexDirection.propTypes,
  ...flexWrap.propTypes,
  ...fontFamily.propTypes,
  ...fontSize.propTypes,
  ...fontWeight.propTypes,
  ...height.propTypes,
  ...justifyContent.propTypes,
  ...justifyItems.propTypes,
  ...justifySelf.propTypes,
  ...left.propTypes,
  ...lineHeight.propTypes,
  ...maxHeight.propTypes,
  ...maxWidth.propTypes,
  ...minHeight.propTypes,
  ...minWidth.propTypes,
  ...opacity.propTypes,
  ...order.propTypes,
  ...overflow.propTypes,
  ...position.propTypes,
  ...right.propTypes,
  ...size.propTypes,
  ...space.propTypes,
  ...textAlign.propTypes,
  ...top.propTypes,
  ...width.propTypes,
  ...zIndex.propTypes
}

export default Box
