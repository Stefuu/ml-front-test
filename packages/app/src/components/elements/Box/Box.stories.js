import { storiesOf } from '@storybook/react'
import { doc } from 'storybook-readme'
import readmeFile from './README.md'

storiesOf('app/components/elements/Box', module)
  .addParameters({ jest: ['Box.test.js'] })
  .add('README', doc(readmeFile))
