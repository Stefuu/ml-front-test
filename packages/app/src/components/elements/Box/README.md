# Box

## Styled div with most properties from styled-system
