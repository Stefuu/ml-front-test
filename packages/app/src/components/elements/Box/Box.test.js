import React from 'react'
import { render } from 'react-testing-library'
import Box from './Box'

describe('Box', () => {
  it('Should render [snapshot]', async () => {
    const { container } = render(<Box />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
