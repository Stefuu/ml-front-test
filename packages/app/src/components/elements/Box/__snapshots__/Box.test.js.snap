// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Should render [snapshot] 1`] = `
<div
  class="sc-bdVaJa fHkBQb"
/>
`;
